### ISSUE TITLE
Swappable batteries not charging.

### ISSUE THREAD

**Issue by Noctifer-de-Mortem** [CleverRaven/Cataclysm-DDA#20286]
I have four swappable battery case with batteries that don;t seem to charge. Their charge has gone down since installed and 10 solar panels later, a v6+truck alternator combo and jumper cables later there still is no charge. Connected car batteries do hold a charge and are a max capacity.

**Comment by kolsurma** [comment]
What do you have attached to the vehicle that is drawing power?

**Comment by Noctifer-de-Mortem** [comment]
But that does not explain the swappable batteries not charging via jumper cables or truck alternator when the other types of batteries do.

**Comment by Coolthulhu** [comment]
Can't reproduce. Could be just that storage batteries charge much slower than the rest.

**Comment by kolsurma** [comment]
4.0L V6 engine with truck alternator used on both sets of tests

|charge from self|empty car battery|empty storage battery -no case| empty storage battery -case|
|-|:-|:-|:-
5 minutes|157|152|154

|charge from other car with shiny jumper cables |empty car battery|empty storage battery -no case| empty storage battery -case|
|-|:-|:-|:-
5 minutes|115|117|115

Everything seems in order to me.

**Comment by Noctifer-de-Mortem** [comment]
Guess I must be doing something wrong then.

**Comment by kevingranade** [comment]
On Feb 17, 2017 5:23 PM, "Noctifer-de-Mortem" <notifications@github.com>
wrote:

Guess I must be doing something wrong then.

It happens to everyone, thanks for reporting it.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/CleverRaven/Cataclysm-DDA/issues/20286#issuecomment-280809486>,
or mute the thread

**Comment by keyspace** [comment]
``` sh
grep -E "(id|name|capacity|[{}])" data/json/items/vehicle/battery.json
```

gives:

``` json
  {
    "id": "battery_car",
    "name": "car battery",
    "name_plural": "car batteries",
    "description": "A 12v lead-acid battery used to power car electrical systems.",
    "capacity": 2500, "//": "58Ah @ 12VDC. Could supply ~700 watts for an hour",
  },
  {
    "id": "battery_motorbike",
    "name": "motorbike battery",
    "name_plural": "motorbike batteries",
    "description": "A 12v lead-acid battery used to power smaller vehicles' electrical systems.",
    "capacity": 500, "//": "12Ah @ 12VDC. Could supply ~140 watts for an hour"
  },
  {
    "id": "medium_storage_battery",
    "name": "medium storage battery",
    "name_plural": "medium storage batteries",
    "capacity": 7000
  },
  {
    "id": "small_storage_battery",
    "name": "small storage battery",
    "name_plural": "small storage batteries",
    "capacity": 500
  },
  {
    "id": "storage_battery",
    "name": "storage battery",
    "name_plural": "storage batteries",
    "capacity": 40000,
  }
```

1% change for a car battery is 25 charge (`battery`). 1% change for storage battery is 400.

**Comment by keyspace** [comment]
@kolsurma solar panels charging during bad weather was a bug. The fix in PR #20278 properly addressed erroneous subtraction of the weather modifier, which is negative at night and positive at day.

I've checked the latter with a simple:

``` c++
107     add_msg( "turn: %f, mod: %f, tick: %f, data: %f",
108              turn.sunlight(), weather_data( wtype ).light_modifier, tick_sunlight, data.sunlight );
```

If anything, that PR fixed a long-standing bug of perceived "solars nerfed", due to them getting charged during night mostly.

Hooray to @UrbanCMC and sign reversal!

Now, it seems that lowering sunlight during bad weather is not applied, but I'll open a separate issue for that.
